packages/integration-sdk-runtime/src/execution/duplicateKeyTracker.ts

```diff
@@ -144,12 +234,38 @@ export type DuplicateEntityReport = {
 function compareEntities(a: Entity, b: Entity): DuplicateEntityReport {
   const aClone = JSON.parse(JSON.stringify(a));
   const bClone = JSON.parse(JSON.stringify(b));
-  aClone._rawData = undefined;
-  bClone._rawData = undefined;
+  delete aClone._rawData;
+  delete bClone._rawData;
+
+  const rawDataMatch = isDeepStrictEqual(a._rawData, b._rawData);
+  const propertiesMatch = isDeepStrictEqual(aClone, bClone);
+
+  let rawDataDiff: ObjectDiff | undefined;
+  if (!rawDataMatch) {
+    try {
+      rawDataDiff = diffObjects(
+        a._rawData?.[0].rawData,
+        b._rawData?.[0].rawData,
+      );
+    } catch (e) {
+      // ignore
+    }
+  }
+
+  let propertiesDiff: ObjectDiff | undefined;
+  if (!propertiesMatch) {
+    try {
+      propertiesDiff = diffObjects(aClone, bClone);
+    } catch (e) {
+      // ignore
```

Comment:
It'd be nice to get a log that the diffObject operation failed.

Reply:
This is hard to do, as we don't pass a logger to the job state code, I could add a property for the error message to the return object of this function, this ends up in a log so it could work, but it's weird. Let me know what you think!